bitcoin/src/blockdata/witness.rs

```diff
@@ -269,17 +267,22 @@ impl Witness {
         self.content[end_varint..end_varint + new_element.len()].copy_from_slice(new_element);
     }
 
+    /// Pushes a [`crate::ecdsa::Signature`] as a new element on the witness (the DER-encoded
+    /// secp256k1::ecdsa::Signature along with the hash type), requires an allocation.
+    pub fn push_signature(&mut self, signature: &crate::ecdsa::Signature) {
+        self.push_bitcoin_signature(&signature.sig.serialize_der(), signature.hash_ty)
+    }
+
     /// Pushes a DER-encoded ECDSA signature with a signature hash type as a new element on the
     /// witness, requires an allocation.
-    pub fn push_bitcoin_signature(&mut self, signature: &ecdsa::SerializedSignature, hash_type: EcdsaSighashType) {
+    pub fn push_bitcoin_signature(&mut self, signature: &secp256k1::ecdsa::SerializedSignature, hash_type: EcdsaSighashType) {
```

Comment:
Seems like it'd be nicer to just change the type? It's confusing why there are two methods with funny name. Or require `T: Into<SerializedSignature>` and impl `From<&Signature> for SerializedSignature`. If we want to support both. (pass-by value is a bit messy but we could fix it with de-monomorphisation and inlining)

Reply:
Cool, I was being a bit conservative and not breaking the current method - I couldn't work out why it was like that in the first place and not just as the new method is (i.e., taking a `bitcoin::ecdsa::Signature`). Will revist using your ideas.

Reply:
`Into` is even backwards-compatible. :) I guess `SerializedSignature` is more efficient if you need to push the same signature multiple times but that should be very rare and optimizing the script to not do that would bring far more cost saving than reducing CPU instructions.